packages/core-test-kit/src/test-stylable-core.ts

```diff
@@ -0,0 +1,66 @@
+import { testInlineExpects } from './inline-expectation';
+import { Stylable, StylableConfig, StylableExports, StylableMeta } from '@stylable/core';
+import { createMemoryFs } from '@file-services/memory';
+import type { IDirectoryContents, IFileSystem } from '@file-services/types';
+import { isAbsolute } from 'path';
+
+export interface TestOptions {
+    entries: string[];
+    stylableConfig: TestStylableConfig;
+}
+
+export type TestStylableConfig = Omit<
+    StylableConfig,
+    'fileSystem' | `projectRoot` | `resolveNamespace`
+> & {
+    filesystem?: IFileSystem;
+    projectRoot?: StylableConfig['projectRoot'];
+    resolveNamespace?: StylableConfig['resolveNamespace'];
+};
+
+export function testStylableCore(
+    input: string | IDirectoryContents,
+    options: Partial<TestOptions> = {}
+) {
+    // infra
+    const fs =
+        options.stylableConfig?.filesystem ||
+        createMemoryFs(typeof input === `string` ? { '/entry.st.css': input } : input);
+    const stylable = Stylable.create({
+        fileSystem: fs,
+        projectRoot: '/',
+        resolveNamespace: (ns) => ns,
+        ...(options.stylableConfig || {}),
+    });
+
+    // collect sheets
+    const allSheets = fs.findFilesSync(`/`, { filterFile: ({ path }) => path.endsWith(`.st.css`) });
```

Comment:
The root will always be `/`?

Reply:
currently supporting in memory fs, if we ever want to run this on real fs then this will need to change

Reply:
So you can add it to the options object, optional `rootDir`option, and the default value would be `/`.

Reply:
Will try

Reply:
might extend in the future, but currently no core tests requires such API